app/main.py

```diff
@@ -2,0 +1,25 @@
+from app.cafe import Cafe
+from app.errors import VaccineError, NotWearingMaskError
+
+
+def go_to_cafe(friends: list, cafe: Cafe) -> str:
+    masks_to_buy = 0
+
+    # Check each friend using try/except
+    for friend in friends:
+        try:
+            # Try to allow the friend to visit the cafe
+            cafe.visit_cafe(friend)
+        except VaccineError:
+            # Catch both NotVaccinatedError and OutdatedVaccineError
+            return "All friends should be vaccinated"
+        except NotWearingMaskError:
+            # Count how many friends are not wearing masks
+            masks_to_buy += 1
+
+    # If any friends don't have masks, return mask buying message
+    if masks_to_buy > 0:
```

Comment:
Code Efficiency: Instead of comparing `masks_to_buy` to 0, you can simply use `if masks_to_buy:` to check if any masks need to be bought.